client/test/components/ContentNavigator/convert.test.ts

```diff
@@ -0,0 +1,466 @@
+  flowData.modifiedTimeStamp = "";
+  flowData.createdBy = "";
+  flowData.modifiedBy = "";
+  const keys = Object.keys(flowData.nodes);
+  for (let i = 0; i < keys.length; i++) {
+    const newKey = "IDSWIMLANE" + i.toString();
+    const oldKey = keys[i];
+    flowData.nodes[newKey] = flowData.nodes[oldKey];
+    delete flowData.nodes[oldKey];
+    flowData.nodes[newKey].id = newKey;
+    flowData.nodes[newKey].dataFlowAndBindings.dataFlow.creationTimeStamp = "";
+    flowData.nodes[newKey].dataFlowAndBindings.dataFlow.modifiedTimeStamp = "";
+    flowData.nodes[newKey].dataFlowAndBindings.dataFlow.createdBy = "";
+    flowData.nodes[newKey].dataFlowAndBindings.dataFlow.modifiedBy = "";
+    const nodeKeys = Object.keys(
+      flowData.nodes[newKey].dataFlowAndBindings.dataFlow.nodes,
+    );
+    for (let j = 0; j < nodeKeys.length; j++) {
+      const newNodeKey = "IDNODE" + i.toString() + j.toString();
+      const oldNodeKey = nodeKeys[j];
+      flowData.nodes[newKey].dataFlowAndBindings.dataFlow.nodes[newNodeKey] =
+        flowData.nodes[newKey].dataFlowAndBindings.dataFlow.nodes[oldNodeKey];
+      delete flowData.nodes[newKey].dataFlowAndBindings.dataFlow.nodes[
+        oldNodeKey
+      ];
+      flowData.nodes[newKey].dataFlowAndBindings.dataFlow.nodes[newNodeKey].id =
+        newNodeKey;
+      flowData.nodes[newKey].dataFlowAndBindings.dataFlow.nodes[
+        newNodeKey
+      ].note.id = "IDNOTE" + i.toString() + j.toString();
+      flowData.nodes[newKey].dataFlowAndBindings.dataFlow.nodes[
+        newNodeKey
+      ].properties.UI_PROP_NODE_DATA_MODIFIED_DATE = "";
+    }
+  }
+  return flowData;
+}
+
+function removeIds(flowData, mode) {
+  if (mode === "Node") {
+    return removeIdsNodes(flowData);
+  }
+  return removeIdsSwimlanes(flowData);
+}
+
+function convertAndRemoveIds(
+  contentSASNotebook,
+  inputName,
+  resultFlowData,
+  mode,
+) {
+  const contentSASNotebookString = JSON.stringify(contentSASNotebook);
+  const flowDataString = convertNotebookToFlow(
+    contentSASNotebookString,
+    inputName,
+    "test.flw",
+  );
+  let flowDataNode = parseFlowData(flowDataString);
+  flowDataNode = removeIds(flowDataNode, mode);
+  // test type of resultFlowData
+  let flowDataResNode = {};
+  if (typeof resultFlowData === "string") {
+    flowDataResNode = parseFlowData(resultFlowData);
+    flowDataResNode = removeIds(flowDataResNode, mode);
+  } else {
+    flowDataResNode = removeIds(resultFlowData, mode);
+  }
+  return [flowDataNode, flowDataResNode];
+}
+
+async function updateWorkspaceSettings(Mode) {
+  const configuration = workspace.getConfiguration(); // Get the workspace configuration
+  await configuration.update("SAS.flowConversionMode", Mode, true);
+}
+
+describe("Notebook Conversion", async () => {
+  it("convert the multi cell sas notebook to node flow", async () => {
+    await updateWorkspaceSettings("Node");
+    const [flowDataNode, flowData1Node] = convertAndRemoveIds(
```

Comment:
nit: (applies to variable names in many places) would it be better to name them with `expected` and `actual` to make them cleaner?

Reply:
done in 4b45ea3cd0c6ebd1a21ab73e963182ae250eb717